xbmc/filesystem/FileCache.cpp

```diff
@@ -605,6 +605,7 @@ int CFileCache::IoControl(EIoControl request, void* param)
   if (request == IOCTRL_CACHE_STATUS)
   {
     SCacheStatus* status = (SCacheStatus*)param;
+    status->maxforward = m_forwardCacheSize;
```

Comment:
Perhaps make this: status->maxforward = m_forwardCacheSize ? m_forwardCacheSize : m_fileSize; ? This allows cache level calculation when disk-cache is used.